OpenEdXMobile/src/main/java/org/edx/mobile/util/DataBindingHelperUtils.kt

```diff
@@ -60,11 +64,13 @@ class DataBindingHelperUtils {
         fun addView(linearLayout: LinearLayout, list: ArrayList<CourseDateBlock>, clickListener: OnDateBlockListener) {
             val inflater: LayoutInflater = linearLayout.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
             if (linearLayout.childCount < 2) {
+                // if all the item has the same date type so it means badge is already added to parent
+                val parentBadgeAdded = hasSameDateTypes(list)
                 list.forEach { item ->
```

Comment:
check if the object is a dummy or not as we are adding today's object manually
i.e, 
`if(item.isDummyObject())`
`       return@forEach` 

we can define `isDummyObject()` method in `CourseDateBlock.kt` as
`fun isDummyObject():Boolean = title.isBlank() && link.isBlank() && linkText.isBlank() && description.isBlank() && isToday()`